src/main/java/thebotfather/task/Task.java

```diff
@@ -0,0 +1,118 @@
+package thebotfather.task;
+
+import thebotfather.util.TheBotFatherException;
+
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+
+/**
+ * Represents a task with a description, completion status, and a type identifier.
+ * This class serves as an abstract base for more specific types of tasks.
+ */
+public abstract class Task {
+
+    /**
+     * Formatter for input date and time strings.
+     */
+    protected static final DateTimeFormatter DATE_STRING_FORMATTER = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm");
+
+    /**
+     * Formatter for printing date and time in a more readable format.
+     */
+    protected static final DateTimeFormatter DATE_STRING_FORMATTER_PRINT = DateTimeFormatter.ofPattern("dd MMM yyyy, HH:mm");
```

Comment:
I think this line exceeds the hard [limit](https://se-education.org/guides/conventions/java/intermediate.html#layout) of 120 characters